frontend/src/pages/ProgressionChecker/GridView/GridView.jsx

```diff
@@ -56,7 +60,9 @@ const GridView = ({ isLoading, structure }) => {
                 newGridLayout[group][subgroup].push({
                   key: courseCode,
                   title: courseData.title,
-                  completed: courseCode in pastCourses,
+                  planned: courseCode in plannerCourses,
+                  past: plannerCourses[courseCode].past,
```

Comment:
Maybe we can remove keeping track of the `past` field here? Don't think it's necessary to keep track of it where we can just calculate on the fly (inside `courseBadge`)?

Not 100% sure about if this is possible tho. Up to you if you want to keep this field in or not. :)